kubetest/util.go

```diff
@@ -63,11 +64,26 @@ func init() {
 }
 
 // Returns $GOPATH/src/k8s.io/...
-func k8s(parts ...string) string {
-	p := []string{os.Getenv("GOPATH"), "src", "k8s.io"}
-	for _, a := range parts {
-		p = append(p, a)
+func k8s(topdir string, parts ...string) string {
+	gopath_list := filepath.SplitList(build.Default.GOPATH)
+	found := false
+	var kubedir string
+	for _, gopath := range gopath_list {
+		kubedir = filepath.Join(gopath, "src", "k8s.io", topdir)
+		if _, err := os.Stat(kubedir); !os.IsNotExist(err) {
```

Comment:
> can there be other errors? like permission error?

Perhaps, but it's really hard to be comprehensive when testing this...

I guess when I wrote this I was thinking of a test in shell that would do `[[ -e "${directory}" ]]` which is essentially what I translated it to...

But perhaps the correct check here is `[[ -d "${directory}" ]]` which would translate to the below?

```golang
if fi, err := os.Stat(kubedir); err != nil && fi.Mode().IsDir() {
```

(Note that this *should* still work on symlinks, since you need [Lstat](https://golang.org/pkg/os/#Lstat) to prevent traversal.)

What do you think?

> probably check err != nil first

`os.IsNotExist` handles `err == nil` correctly (according to the [example](https://golang.org/pkg/os/#IsNotExist) in its documentation, and I also double checked it.)